userland/Program.mk

```diff
@@ -3,18 +3,46 @@
 $(call check_defined, BUILDDIR)
 $(call check_defined, PACKAGE_NAME)
 
-APP_TOCKLOADER ?= tockloader
+TOCKLOADER ?= tockloader
+OPENOCD ?= openocd
 
 # Upload programs over UART with tockloader
 ifdef PORT
   TOCKLOADER_GENERAL_FLAGS += --port $(PORT)
 endif
 
+#
+# Setup specific commands for each board.
+#
+ifeq ("$(TOCK_BOARD)","hail")
+PROGRAM = $(TOCKLOADER) $(TOCKLOADER_GENERAL_FLAGS) install $<
+FLASH = $(TOCKLOADER) $(TOCKLOADER_GENERAL_FLAGS) install --jtag $<
+
+else ifeq ("$(TOCK_BOARD)","imix")
+# Change program region offset
+TOCKLOADER_INSTALL_FLAGS += --app-address 0x40000
+PROGRAM = $(TOCKLOADER) $(TOCKLOADER_GENERAL_FLAGS) install $(TOCKLOADER_INSTALL_FLAGS) $<
+FLASH = $(TOCKLOADER) $(TOCKLOADER_GENERAL_FLAGS) install $(TOCKLOADER_INSTALL_FLAGS) --jtag $<
+
+else ifeq ("$(TOCK_BOARD)","ek-tm4c1294xl")
+FLASH = $(OPENOCD) -c "source [find board/ek-tm4c1294xl.cfg]; init; reset halt; flash write_image erase $< 0x00020000 bin; reset; shutdown"
+
+else ifeq ("$(TOCK_BOARD)","nrf51dk")
+FLASH = $(TOCK_USERLAND_BASE_DIR)/tools/program/nrf51dk.py $<
```

Comment:
Should we remove `tools/program/nrf5x.py` and use tockloader instead?

Reply:
We absolutely should support the nRF51 and 52 in tockloader. Does that support exist right now?

Reply:
I use tockloader for the nrf52. `tockloader install --jlink --board nrf52dk`

Reply:
Yepp, we support both but I think we should aim to use tockloader as much as possible and reduce the amount of board-specific scripts if that is possible!

However ```make TOCK_BOARD=nrf52dk``` is using userland/tools/program/nrf52dk.py under the hood!

Reply:
Having chatted with Brad about this a little out-of-band, I think we should move the nRF platforms over to Tockloader by default as a separate pull request. It's a good thing to do, but not really part of this change.